app/controllers/bids_controller.rb

```diff
@@ -0,0 +1,64 @@
+class BidsController < ApplicationController
+  before_action :authenticate_any!
+  before_action :ensure_business_user! ,except: :index
+  before_action :ensure_my_job!, only: :index
+  before_action :ensure_one_bid_by_one_user_for_one_job!, only: :new
+  before_action :is_owner_of_bid?, only: :destroy
+
+  def index
+    @bids = Bid.where(job_id: params[:job_id]).order("created_at DESC")
+  end
+
+  def new
+    @bid = Bid.new
+  end
+
+  def create
+    @job = Job.find(params[:job_id])
+    @bid = @job.bids.new(bids_params)
+    if @bid.save
+      redirect_to root_path
+    else
+      render :new, status: :inprocessable_entity
+    end
+  end
+
+  def destroy
+    @bid = Bid.find(params[:id])
+    @bid.destroy
+    flash[:success] = "Bid has been sucessfully deleted"
+    redirect_to job_bids_path(job_id: params[:job_id])
+  end
+
+  private
+
+  def bids_params
+    params.require(:bid).permit(:cover_letter, :price).merge(bidder_id: current_user.id, bidder_type: current_user.class.name)
+  end
+
+  def ensure_business_user!
+    return unless business_user_signed_in?
+    flash[:alert] = "Your Account type is business user you can't bid to jobs"
+    redirect_to root_path
+  end
+
+  def ensure_my_job!
```

Comment:
We might ensure that only  business user can't see other jobs posted by other SB.